### PR TITLE
[AMSFT102020]issaGruber

### PR DESCRIPTION
**Pull request by GruberI**
Finished iterations 1-6. Tried bonus but got lost trying to get search to work. Would love to get help figuring it out. 

Also, getting some random beers from db like Emma, Emad, IreneTest. Assume it's students adding?

### CODE REVIEW

File: src/components/RandomBeer/RandomBeer.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import Header from '../Header/Header'
+import './RandomBeer.css'
+
+const RandomBeer = () => {
+    const [randomBeer, setRandomBeer] = useState({})
+
+    useEffect(() => {
+        axios.get("https://ih-beers-api2.herokuapp.com/beers/random")
+        .then((beersResults) => {
+            console.log(beersResults)
```
Comment (LloydC) [inline src/components/RandomBeer/RandomBeer.js:12]: Remember to remove unnecessary `console.logs` after you're done testing.

File: src/components/NewBeer/NewBeer.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    const inputChangeHandler = (event) => {
+        let {name, value} = event.target;
+        setBeerValues({...beerValues, [name]: value});
+    }
+
+    const addBeer = (event) => {
+        event.preventDefault();
+
+        axios.post('https://ih-beers-api2.herokuapp.com/beers/new', beerValues)
+        .then()
```
Comment (LloydC) [inline src/components/NewBeer/NewBeer.js:28]: To improve the User Experience, you could reset the form to it's initial state.
```
axios.post('https://ih-beers-api2.herokuapp.com/beers/new', beerValues)
.then(() =>{
 setBeerValues(initialState);
})

File: src/components/NewBeer/NewBeer.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        .catch((error) => error);
```
Comment (LloydC) [inline src/components/NewBeer/NewBeer.js:29]: You forgot to add the `console.log(error)`.

File: src/components/BeersList/BeersList.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    const filterFoodsHandler = ((searchInput) => {
+        axios.get("https://ih-beers-api2.herokuapp.com/beers/search?q={query}")
+        .then((searchResults) => {
+            searchResults.filter((beer) =>{
+                beer.name.inclludes(searchInput)
+            })
+            setBeersList(searchResults)
+        })
+        .catch((err) => {console.error(err)})
+    },[])
```
Comment (LloydC) [inline src/components/BeersList/BeersList.js:29]: Your function doesn't need an empty array as second argument.

File: src/components/BeersList/BeersList.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    useEffect(() => {
+        axios.get("https://ih-beers-api2.herokuapp.com/beers")
+        .then((beersResults) => {
+            console.log(beersResults)
+            setBeersList(beersResults.data)
+        })
+        .catch((err) => {console.error(err)})
+    },[])
+
```
Comment (LloydC) [inline src/components/BeersList/BeersList.js:19]: Good effort, here's a working `filterFoodsHandler`
```
const filterFoodsHandler = ((searchInput) => {
        axios.get(`https://ih-beers-api2.herokuapp.com/beers/search?q=${searchInput}`)
        .then((searchResults) => {
            setBeersList(searchResults.data)
        })
        .catch((err) => {console.error(err)})
    })

File: src/components/BeerSearch/BeerSearch.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React, {useState} from 'react';
+import axios from 'axios';
+import './BeerSearch.css'
+
+
+const BeerSearch = () => {
+    const [beerSearch, setBeerSearch] = useState([])
+
```
Comment (LloydC) [inline src/components/BeerSearch/BeerSearch.js:8]: To get your `handleSearchInput` function to work, you must use the search function you defined in `Beerlist.js` from the `props` and pass it `event.target.value`.
```
const {handleFilterSearch} = props;

    const handleSearchInput = (event) => {
        let initialState = event.target.value;
        handleFilterSearch(initialState)
    }

File: src/App.js [hunk 7]
```diff
@@ -1,25 +1,26 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
```
Comment (LloydC) [inline src/App.js:3]: You should have extracted [`Switch`](https://reactrouter.com/web/api/Switch) as well.

File: src/App.js [hunk 8]
```diff
@@ -1,25 +1,26 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Router>
+      <div>
```
Comment (LloydC) [inline src/App.js:16]: This is where you need to add the `<Switch>` component.

**Comment by LloydC** [comment]
Correct assumption. I added a workable solution to Iteration 7 in my comments. 

It should be straightforward but feel free to reach out if you have any questions about it.

Good progress there!

**Comment by GruberI** [comment]
Thanks Lloyd I will try this out today and implement your suggestions!

**Pull request closed without merging**